src/vtvalidate.cpp

```diff
@@ -5,55 +5,83 @@
-            throw std::runtime_error("Uh oh, this should never happen"); // NOLINT
+struct geom_handler {
+
+    void points_begin(uint32_t /*dummy*/) {}
+
+    void points_point(const vtzero::point /*dummy*/) {}
+
+    void points_end() const noexcept {}
+
+    void linestring_begin(uint32_t /*dummy*/) {}
+
+    void linestring_point(const vtzero::point /*dummy*/) {}
+
+    void linestring_end() const noexcept {}
+
+    void ring_begin(uint32_t /*dummy*/) {}
+
+    void ring_point(const vtzero::point /*dummy*/) {}
+
+    void ring_end(vtzero::ring_type /*dummy*/) const noexcept {}
+
+}; // struct geom_handler
+
+std::string parseTile(vtzero::data_view const& buffer) {
+    std::string result;
+
+    vtzero::vector_tile tile{buffer};
+    try {
+        while (auto layer = tile.next_layer()) {
+            while (auto feature = layer.next_feature()) {
+                // Detect geomtype of feature and decode
+                geom_handler handler;
+                vtzero::decode_geometry(feature.geometry(), handler);
+
+                feature.reset_property(); // restart the property iterator from the beginning
+                feature.for_each_property([&](vtzero::property const& p) {
+                    p.key();
+                    auto value = p.value();
+                    switch (value.type()) {
+                    case vtzero::property_value_type::string_value:
+                        value.string_value();
+                        break;
+                    case vtzero::property_value_type::float_value:
+                        value.float_value();
+                        break;
+                    case vtzero::property_value_type::double_value:
+                        value.double_value();
+                        break;
+                    case vtzero::property_value_type::int_value:
+                        value.int_value();
+                        break;
+                    case vtzero::property_value_type::uint_value:
+                        value.uint_value();
+                        break;
+                    case vtzero::property_value_type::sint_value:
+                        value.sint_value();
+                        break;
+                    case vtzero::property_value_type::bool_value:
+                        value.bool_value();
+                        break;
+                    default:
+                        throw std::runtime_error("Invalid property value type");
+                    }
+                    return true; // continue to next property
+
+                });
+            }
         }
+    } catch (std::exception const& ex) {
+        result = ex.what();
+        return result;
     }
 
-    std::string result = "...threads are busy async bees...hello world";
-
-    if (louder) {
-        result += "!!!!";
-    }
-
+    result = "true";
```

Comment:
@GretaCB how about we only set the `result` for errors? Then if it is empty we know the tile was "valid".